packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -499,7 +503,10 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
                     }
                 };
 
-                const disposable = this.term.onData(sendData);
+                const disposable = new DisposableCollection;
+                disposable.push(this.term.onData(sendData));
+                disposable.push(this.term.onBinary(sendData));
```

Comment:
It might work for now, but I'd like someone to understand the implications of doing this?

- [Xterm.js mentions how this binary data should be consumed](https://xtermjs.org/docs/guides/encoding/#output).
- [VS Code seems to follow it](https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/terminalProcess.ts#L413).

Either we can actually just shove it that way, same as the regular `onData` event, or we're missing some processing?

Reply:
In theia cases, both node-pty in Macos and Windows use Nodejs-PipeSocket as ipc. And PipeSocket were initialized by encoding=undefined, which mean socket treat input flow as utf-8 string. String data would be convert to Buffer after write to Nodejs-Socket by Buffer.from(chunk,  'utf-8').
Binary string is equal to latin string. Every code unit tread as a latin character (0 ~ 255).So It's safe apply  Buffer.from(chunk, 'utf-8') to binary-string because latin code point was included by UTF-8 charset.
The problem of padding for convert latin string to UTF-8 string, actually happend in convert a Buffer to string. That's problem of LANG or LC_ALL environment variable of shell. So you can temporary ignore these case.

Reply:
The second argument of Buffer.from is meaning "How Buffer treat every string code point and save them in UInt8Array?".Both latin string and utf-8 string use 1 byte.